packages/lib/server/getServerErrorFromUnknown.ts

```diff
@@ -36,11 +38,37 @@ function parseZodErrorIssues(issues: ZodIssue[]): string {
     .join("; ");
 }
 
+const twilioErrorLogger = logger.getSubLogger({ prefix: ["TwilioError"] });
+
 export function getServerErrorFromUnknown(cause: unknown): HttpError {
   if (cause instanceof TRPCError) {
     const statusCode = getHTTPStatusCodeFromError(cause);
     return new HttpError({ statusCode, message: cause.message });
   }
+  if (isTwilioError(cause)) {
+    twilioErrorLogger.error("Twilio error occurred:", {
+      status: cause.status,
+      message: cause.message,
+      code: cause.code,
+      cause: cause,
+    });
```

Comment:
@keithwillcode
- Main consumer of `getServerErrorFromUnknown` is `defaultResponder` and `defaultResponderAppDir`, both of which are wrappers of many web API endpoints including booking ones. 
- `defaultResponder` and `defaultResponderAppDir` both only log errors >= 500 or  < 400.
- That's why I added a manual log here. But I wonder if we should just propagate `cause.status` as status code, or we should make it 202 for logging as well if the status code is not 400.